### ISSUE TITLE
Error running nafToAbsa2015

### ISSUE THREAD

**Issue by haohao-hu** [ragerri/ixa-pipe-convert#1]
I tried the following and got the nullpointerexception. By the way, when building this app, maven reported no package 0.0.3 found in maven2/eus/ixa/ixa-pipe-ml/, so I switched to 0.0.2 instead

`$ java -jar target/ixa-pipe-convert-0.3.0.jar --nafToAbsa2015 OTE_absa2015RTrain                                                                                                       _YRRCmodelBROWNW2V.naf > OTE_absa2015RTrain_YRRCmodelBROWNW2V.xml
CLI options: Namespace(yelpGetText=null, normalizePennTreebank=null, absa2015Tex                                                                                                       t=null, absa2015ToCoNLL2002=null, serializeClarkCluster=null, removeEntities=nul                                                                                                       l, serializeEntityDictionary=null, ancora2treebank=null, printNER=null, serializ                                                                                                       eLemmaDictionary=null, absa2014ToCoNLL2002=null, addLemmaDict2POSDict=null, prin                                                                                                       tNED=null, treebank2WordPos=null, createPOSDictionary=null, neTypes=null, dsrcTo                                                                                                       CoNLL02=null, nafToAbsa2015=OTE_absa2015RTrain_YRRCmodelBROWNW2V.naf, lowercase=                                                                                                       false, brownClean=null, absa2015ToNAF=null, nafToAbsa2014=null, nafToCoNLL03=nul                                                                                                       l, treebank2tokens=null, parseToChunks=null, nafToCoNLL02=null, serializeBrownCl                                                                                                       uster=null, filterNameTypes=null, parseToTabulated=null, createMonosemicDictiona                                                                                                       ry=null, absa2015ToWFs=null)
Exception in thread "main" java.lang.NullPointerException
        at eus.ixa.ixa.pipe.convert.AbsaSemEval.getReviewIdsFromXpathAttribute(A                                                                                                       bsaSemEval.java:434)
        at eus.ixa.ixa.pipe.convert.AbsaSemEval.nafToAbsa2015(AbsaSemEval.java:3                                                                                                       50)
        at eus.ixa.ixa.pipe.convert.CLI.main(CLI.java:235)
`

**Comment by ragerri** [comment]
Hello,

Install locally ixa pipe ml (mvn clean install) and then compile ixa-pipe-convert. Having said that, can you show me the NAF file you are trying to convert?

**Comment by haohao-hu** [comment]
Thanks, I successfully installed 0.3.0 yesterday. But I still got the same error when trying to convert naf to ABSA2015 format. Github dont support Naf file, so I changed it to txt file and uploaded.

[OTE_absa2015RTrain_YRRCmodelBROWNW2V.txt](https://github.com/ragerri/ixa-pipe-convert/files/948632/OTE_absa2015RTrain_YRRCmodelBROWNW2V.txt)

**Comment by ragerri** [comment]
And the resultant XML works fine with the ABSA evaluation script? Because it should not. What you need to do is:
1. java -jar ixa-pipe-convert-0.3.0.jar --absa2015ToWFs $ABSA_TEST to obtain the NAF WFs with the document ids included
2. Run the pos tagger and lemmatizer the way you already do
3. Run ixa-pipe-opinion-1.0.0-exec.jar
4. Run --nafToAbsa2015 to obtain the ABSA formatted XML file.

HTH,

R

**Comment by haohao-hu** [comment]
I tried the above instructions, but  got the following error:
`$ java -jar target/ixa-pipe-convert-0.3.0.jar --nafToAbsa2015 OTE2015REST_TEST-BrownYRC1000-W2VYRC1000-window2_2-1415combined.naf > OTE2015REST_TEST-BrownYRC1000-W2VYRC1000-window2_2-1415combined.xml
CLI options: Namespace(yelpGetText=null, normalizePennTreebank=null, absa2015Text=null, absa2015ToCoNLL2002=null, serializeClarkCluster=null, removeEntities=null, serializeEntityDictionary=null, ancora2treebank=null, printNER=null, serializeLemmaDictionary=null, absa2014ToCoNLL2002=null, addLemmaDict2POSDict=null, printNED=null, treebank2WordPos=null, createPOSDictionary=null, neTypes=null, dsrcToCoNLL02=null, nafToAbsa2015=OTE2015REST_TEST-BrownYRC1000-W2VYRC1000-window2_2-1415combined.naf, lowercase=false, brownClean=null, absa2015ToNAF=null, nafToAbsa2014=null, nafToCoNLL03=null, treebank2tokens=null, parseToChunks=null, nafToCoNLL02=null, serializeBrownCluster=null, filterNameTypes=null, parseToTabulated=null, createMonosemicDictionary=null, absa2015ToWFs=null)
Exception in thread "main" java.lang.NullPointerException
        at eus.ixa.ixa.pipe.convert.AbsaSemEval.nafToAbsa2015(AbsaSemEval.java:383)
        at eus.ixa.ixa.pipe.convert.CLI.main(CLI.java:235)
`

**Comment by haohao-hu** [comment]
I tried commented out the line 383 and category set to "na":
`     //String category = opinion.getOpinionExpression().getSentimentProductFeature();
            String targetString="na";
            targetString = opinion.getStr();

            int fromOffset = opinion.getOpinionTarget().getTerms().get(0).getWFs().get(0).getOffset();
            List<WF> targetWFs = opinion.getOpinionTarget().getTerms().get(opinion.getOpinionTarget().getTerms().size() -1).getWFs();
            int toOffset = targetWFs.get(targetWFs.size() -1).getOffset() + targetWFs.get(targetWFs.size() -1).getLength();
            opinionElem.setAttribute("target", targetString);
            opinionElem.setAttribute("category", "na");`
and I got the following error:
`$ java -jar target/ixa-pipe-convert-0.3.0.jar --nafToAbsa2015 OTE2015REST_TEST-BrownYRC1000-W2VYRC1000-window2_2-1415combined.naf > OTE2015REST_TEST-BrownYRC1000-W2VYRC1000-window2_2-1415combined.xml
CLI options: Namespace(yelpGetText=null, normalizePennTreebank=null, absa2015Text=null, absa2015ToCoNLL2002=null, serializeClarkCluster=null, removeEntities=null, serializeEntityDictionary=null, ancora2treebank=null, printNER=null, serializeLemmaDictionary=null, absa2014ToCoNLL2002=null, addLemmaDict2POSDict=null, printNED=null, treebank2WordPos=null, createPOSDictionary=null, neTypes=null, dsrcToCoNLL02=null, nafToAbsa2015=OTE2015REST_TEST-BrownYRC1000-W2VYRC1000-window2_2-1415combined.naf, lowercase=false, brownClean=null, absa2015ToNAF=null, nafToAbsa2014=null, nafToCoNLL03=null, treebank2tokens=null, parseToChunks=null, nafToCoNLL02=null, serializeBrownCluster=null, filterNameTypes=null, parseToTabulated=null, createMonosemicDictionary=null, absa2015ToWFs=null)
Exception in thread "main" java.lang.NullPointerException
        at ixa.kaflib.Opinion.getStr(Opinion.java:384)
        at eus.ixa.ixa.pipe.convert.AbsaSemEval.nafToAbsa2015(AbsaSemEval.java:385)
        at eus.ixa.ixa.pipe.convert.CLI.main(CLI.java:235)
`

**Comment by ragerri** [comment]
Send the generated NAF using ixa-pipe-convert, please.

**Comment by haohao-hu** [comment]
[OTE2015REST_TEST-BrownYRC1000-W2VYRC1000-window2_2-1415combined.txt](https://github.com/ragerri/ixa-pipe-convert/files/953548/OTE2015REST_TEST-BrownYRC1000-W2VYRC1000-window2_2-1415combined.txt)

**Comment by ragerri** [comment]
You did not follow all the 4 steps, the targets were annotated using ixa-pipe-nerc. The conversion scripts work with the models trained with ixa-pipe-ml (sequenceTrainer -p parametersFile.properties) and ixa-pipe-opinion.

The opinion NAF layer generated by ixa-pipe-opinion is different to that generated by ixa-pipe-nerc. That is the problem.

HTH

**Comment by haohao-hu** [comment]
Ok, thanks. I will try that

**Comment by ragerri** [comment]
Another clarification (ixa-pipe-ml and ixa-pipe-convert are still under development and documentation is scarce):

The brown clusters you use need to be serialized with ixa-pipe-convert --serializeBrownCluster clusterfile and then pass that .gz cluster lexicon to the BrownClusterFeatures as you do now.

The same with Clark and Word2vec clusters.

HTH

**Comment by haohao-hu** [comment]
Ok, thanks for clarifying.  I will try it

**Comment by haohao-hu** [comment]
But where can I find ixa-pipe-opinion, the page for ixa-pipe-opinion on Github is actually ixa-pipe-nerc: https://github.com/ixa-ehu/ixa-pipe-opinion

**Comment by ragerri** [comment]
the README is the same, I have not updated it yet, but the repo and code is different. Just clone that repo and use it ixa-pipe-opinion ote model.bin

**Comment by haohao-hu** [comment]
Ok, thanks. I've trained models using ixa-pipe-ml, and i am going to use  models  in OTE

**Comment by haohao-hu** [comment]
Finally, it worked. I successfully evaluated my OTE result. Thanks again.

**Comment by haohao-hu** [comment]
`$ cat ABSA15_Restaurants_Test.naf | java -jar ixa-pipe-pos/target/ixa-pipe-pos-1.5.1-exec.jar tag -m ixa-pipe-pos/morph-models-1.5.0/en/en-pos-perceptron-autodict01-conll09.bin -lm ixa-pipe-pos/morph-models-1.5.0/en/en-lemma-perceptron-conll09.bin | java -jar target/ixa-pipe-opinion-1.0.0-exec.jar ote -m OTEmodels/ote-2015-restaurants-Agerri.bin > OTE_15RTest_ote-2015-restaurants-Agerri.naf
CLI options: Namespace(clearFeatures=no, model=OTEmodels/ote-2015-restaurants-Agerri.bin, language=null, outputFormat=naf)
CLI options: Namespace(allMorphology=false, lemmatizerModel=ixa-pipe-pos/morph-models-1.5.0/en/en-lemma-perceptron-conll09.bin, beamSize=3, dictag=false, model=ixa-pipe-pos/morph-models-1.5.0/en/en-pos-perceptron-autodict01-conll09.bin, language=null, multiwords=false, outputFormat=naf)
ixa-pipe-pos model loaded in: 1099 miliseconds ... [DONE]
ixa-pipe-lemma model loaded in: 140 miliseconds ... [DONE]
Exception in thread "main" opennlp.tools.util.ext.ExtensionNotLoadedException: Unable to find implementation for opennlp.tools.util.model.ArtifactSerializer, the class or service eus.ixa.ixa.pipe.nerc.dict.ClarkCluster$ClarkClusterSerializer could not be located!
        at opennlp.tools.util.ext.ExtensionLoader.instantiateExtension(ExtensionLoader.java:122)
        at opennlp.tools.util.model.BaseModel.finishLoadingArtifacts(BaseModel.java:323)
        at opennlp.tools.util.model.BaseModel.loadModel(BaseModel.java:256)
        at opennlp.tools.util.model.BaseModel.<init>(BaseModel.java:179)
        at eus.ixa.ixa.pipe.ml.sequence.SequenceLabelerModel.<init>(SequenceLabelerModel.java:137)
        at eus.ixa.ixa.pipe.ml.StatisticalSequenceLabeler.loadModel(StatisticalSequenceLabeler.java:280)
        at eus.ixa.ixa.pipe.ml.StatisticalSequenceLabeler.<init>(StatisticalSequenceLabeler.java:101)
        at eus.ixa.ixa.pipe.opinion.Annotate.<init>(Annotate.java:64)
        at eus.ixa.ixa.pipe.opinion.CLI.extractOte(CLI.java:193)
        at eus.ixa.ixa.pipe.opinion.CLI.parseCLI(CLI.java:140)
        at eus.ixa.ixa.pipe.opinion.CLI.main(CLI.java:123)
`
Dear Dr. Agerri, I tried to use your OTE model provided in https://github.com/ixa-ehu/ixa-pipe-opinion#ote-models. But I encounted this error. Is it because the model was trained using ixa-pipe-nerc?
